CIP-meta-assets (ERC20-like assets)/README.md

```diff
@@ -0,0 +1,428 @@
+
+This done to facilitate the query of the user's accounts in the offchain.
+
+The standard doesn't include a specification on the policy in case of burning but it is suggested to always allow for the assets burn.
+
+the data needed form the `ScriptContext` includes teh following fields:
+
+- `inputs`
+- `output`
+- `mint`
+
+used as follows:
+
+- all the transaction inputs MUST NOT include any tokens having the same currency symbol of the `accountFactory` minting policy;
+optionally a specific implementation might require a fixed number of inputs (eg. a single input) for performance reasons.
+
+- when minting assets only a single token should be minted (indipendently from the number of inputs)
+- the only asset minted MUST have an unique asset name (we suggest using the output of `[(builtin sha2_256) [(builtin serialiseData) fstUtxoRef]]` where `fstUtxoRef` is the utxo reference (or `PTxOutRef`) of the very first input of the transaction)
+
+- the asset minted MUST be included in an output of the transaction being validated going to the address as specified above (payment credentials that are the hard-coded `accountManager` validator, and stake credtentials of the user if any).
+
+- the output going to the `accountManager` validator MUST implement the following checks (in addition to the presence of the minted asset):
+    - the address' payment credential MUST have `ScriptCredential` constructor (constructor index `1`) and validator hash MUST equal the hard coded `accountManager` hash
+    - the value attached to the UTxO must only have 2 entries (ADA and minted token) to prevent DoS by token spam.
+    - the output datum MUST:
+        - be constructed using the `InlineDatum` consturctor (consturctor index `2`)
+        - have datum value with the [`Account`](#Account) structure (explained below) with the foeeowing fields:
+            - amount: `0`
+            - currencySym: currency symbol of the `accountFactory` minting policy
+            - credentials: the first resolved input address credentials (both public key hash and validator hash supported)
+            - state: no restrictions (up to the specific implementaiton).
+
+optionally a specific implementation might require a fixed number of outputs (eg. a single output) for performance reasons.
+
+### `accountManager` (spending validator)
+
+The `accountManager` contract is responsabile of managing exsiting accounts and their balances.
+
+This is done using some standard redeemers and optionally implementation specfic ones.
+
+### `Account`
+
+The `Account` data type is used as the `accountManager` datum; and is defined as follows:
+
+```ts
+const Account = pstruct({
+    Account: {
+        amount: int,
+        credentials: PCredential.type,
+        currencySym: PCurrencySymbol.type,
+        state: data
+    }
+});
+```
+
+### `AccounManagerRedeemer`
```

Comment:
Should this be called `AccountManagerRedeemer`? Looks like a typo, but was used consistently below.

Reply:
yes it should, is a typo copied and pasted

Reply:
fixed